src/vs/platform/uriLabel/common/uriLabel.ts

```diff
@@ -41,18 +46,21 @@ export class UriLabelService implements IUriLabelService {
 
 	private readonly formaters = new Map<string, UriLabelRules>();
 	private readonly _onDidRegisterFormater = new Emitter<{ scheme: string, formater: UriLabelRules }>();
+	private contextService: IWorkspaceContextService;
 
 	constructor(
 		@IEnvironmentService private environmentService: IEnvironmentService,
-		@IWorkspaceContextService private contextService: IWorkspaceContextService
 	) { }
 
-
 	get onDidRegisterFormater(): Event<{ scheme: string, formater: UriLabelRules }> {
 		return this._onDidRegisterFormater.event;
 	}
 
-	getLabel(resource: URI, relative: boolean, forceNoTildify?: boolean): string {
+	acquireContextService(contextService: IWorkspaceContextService): void {
```

Comment:
@isidorn this is not something we should push because it violates our service injection model. possible solutions:
* extract functionality into a third service that both depend on
* implement the 2 services in one place (e.g. context service and configuration service do this)

Reply:
Configuration service is already have this issue because it is the first service that needs to be created. It cannot be taken as example and follow that pattern for other services.

I see the root cause of this issue is that context service depending on IURILabelService. Due to the above reason, we have to avoid Configuration Service depending on other services as much as possible. 

I think `IWorkspace` object should not have a name property and the label service should provide it. Whoever wants a label for workspace should ask IURILabelService. Removing the name from `IWorkspace` will remove the dependency on `IURILabelService`